metricsaggregator/src/main/java/io/dockstore/metricsaggregator/MetricsAggregatorAthenaClient.java

```diff
@@ -0,0 +1,270 @@
+import io.dockstore.openapi.client.model.SourceControlBean;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.athena.AthenaClient;
+import software.amazon.awssdk.services.athena.model.ColumnInfo;
+import software.amazon.awssdk.services.athena.model.Datum;
+import software.amazon.awssdk.services.athena.model.GetQueryResultsResponse;
+import software.amazon.awssdk.services.athena.model.Row;
+import software.amazon.awssdk.services.athena.paginators.GetQueryResultsIterable;
+
+/**
+ * A class that aggregates metrics using AWS Athena.
+ */
+public class MetricsAggregatorAthenaClient {
+    private static final Logger LOG = LoggerFactory.getLogger(MetricsAggregatorAthenaClient.class);
+    private final AtomicInteger numberOfDirectoriesProcessed = new AtomicInteger(0);
+    private final ExecutionStatusAthenaAggregator executionStatusAggregator = new ExecutionStatusAthenaAggregator();
+
+    private final String metricsBucketName;
+    private final String outputS3Bucket;
+    private final String databaseName;
+    private final String tableName;
+    private final AthenaClient athenaClient;
+    private final MetadataApi metadataApi;
+
+    public MetricsAggregatorAthenaClient(MetricsAggregatorConfig config) {
+        this.metricsBucketName = config.getS3Config().bucket();
+        this.outputS3Bucket = config.getAthenaConfig().outputS3Bucket();
+
+        final String underscoredMetricsBucketName = metricsBucketName.replace("-", "_"); // The metrics bucket name is usually in the form of "env-dockstore-metrics-data"
+        this.databaseName = underscoredMetricsBucketName + "_database";
+        this.tableName = underscoredMetricsBucketName + "_table";
+        this.athenaClient = createAthenaClient();
+        this.metadataApi = new MetadataApi(setupApiClient(config.getDockstoreConfig().serverUrl())); // Anonymous client
+
+        createDatabase();
+        createTable();
+    }
+
+    /**
+     * Aggregate metrics using AWS Athena for the list of S3 directories and posts them to Dockstore.
+     * @param s3DirectoriesToAggregate
+     * @param extendedGa4GhApi
+     * @param skipPostingToDockstore
+     */
+    public void aggregateMetrics(List<S3DirectoryInfo> s3DirectoriesToAggregate, ExtendedGa4GhApi extendedGa4GhApi, boolean skipPostingToDockstore) {
+        // Aggregate metrics for each directory
+        s3DirectoriesToAggregate.stream().parallel().forEach(s3DirectoryInfo -> {
+            Map<String, Metrics> platformToMetrics = getAggregatedMetricsForPlatforms(s3DirectoryInfo);
+            if (platformToMetrics.isEmpty()) {
+                LOG.error("No metrics were aggregated");
```

Comment:
Message could be a little difficult to understand since it's in `parallel()` 
Maybe add some info from `s3DirectoryInfo`